src/client/opamSolution.ml

```diff
@@ -138,6 +138,36 @@ let atom_of_name name =
   name, None
 *)
 
+let did_you_mean st atoms = 
+  let open OpamPackage.Set.Op in 
+  let packages =
+    OpamFormula.packages_of_atoms ~disj:false
+      (st.packages ++ st.installed) atoms
+  in
+  let choices name = 
+    let dict = fun yield -> List.iter yield 
+        (OpamPackage.Set.to_list (st.packages ++ st.installed) |> 
+         List.map (fun p -> OpamPackage.name p |> 
+                            OpamPackage.Name.to_string)) in 
+    OpamCompat.String.spellcheck dict name 
+  in
+  let _, missing_atoms =
+    List.partition (fun (n,_) -> OpamPackage.has_name packages n) atoms
+  in
+  let choices = 
+    List.fold_left (fun acc ma -> 
+        choices (OpamFormula.short_string_of_atom ma) ::acc)
+      [] missing_atoms |> List.concat |> List.sort_uniq String.compare 
```

Comment:
I think the order given to the function should be kept
```suggestion
      [] missing_atoms |> List.concat
```

Reply:
Right, did that to preserve order. However I was using this more for the `uniq` part because the result had multiple suggestion of the same package (per version). 
Had to transform `package_set` into `OpamPackage.Name.Set.t`